### PR TITLE
Controller Extension: Prevent crashes on Amazon FireTV, fixes #5277

### LINKED ISSUE #5277: Controllers on AmazonFireTV: RuntimeException
#### Issue details
I have several crashes from Amazon FireTV (AFTT) sticks when players connect a controller.

I have no idea what is causing this problem, however I wanted to add it here because it is not a one-time problem. I have this crash stacktrace once a day in my crash logs so it is perhaps a good idea to add a catch phrase.

#### Reproduction steps/code
Unknown, probably just connecting a game controller to a Fire TV stick. I was not able to reproduce the problem, if I use a controller on my FireTV stick it works well.

#### Version of LibGDX and/or relevant dependencies
gdx-controllers.

#### Stacktrace
```
java.lang.RuntimeException: Could not get input device information.
	at android.hardware.input.InputManager.getInputDevice(InputManager.java:228)
	at android.view.InputDevice.getDevice(InputDevice.java:410)
	at com.badlogic.gdx.controllers.android.AndroidControllers.addController(Unknown Source)
	at com.badlogic.gdx.controllers.android.ControllerLifeCycleListener.onInputDeviceAdded(Unknown Source)
	at android.hardware.input.InputManager$InputDeviceListenerDelegate.handleMessage(InputManager.java:866)
	at android.os.Handler.dispatchMessage(Handler.java:102)
	at android.os.Looper.loop(Looper.java:135)
	at android.app.ActivityThread.main(ActivityThread.java:5483)
	at java.lang.reflect.Method.invoke(Native Method)
	at java.lang.reflect.Method.invoke(Method.java:372)
	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:984)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:779)
Caused by: android.os.TransactionTooLargeException
	at android.os.BinderProxy.transactNative(Native Method)
	at android.os.BinderProxy.transact(Binder.java:496)
	at android.hardware.input.IInputManager$Stub$Proxy.getInputDevice(IInputManager.java:333)
	at android.hardware.input.InputManager.getInputDevice(InputManager.java:226)
	... 11 more
```

#### Please select the affected platforms
- [X] Android

### PR DESCRIPTION
**Pull request by MrStahlfelge**

### CODE REVIEW

**Comment by PokeMMO** [comment]
Can you add an Gdx.app.log call with the throwable? It would be useful to know why a controller isn't being picked up, even if we can't use it.

**Comment by MrStahlfelge** [comment]
Thought about it, but it is most likely that the input device causing the problem is _not_ a game controller but anything else. The crash is thrown by getDevice, so before the check if it is a game controller is done.

File: extensions/gdx-controllers/gdx-controllers-android/src/com/badlogic/gdx/controllers/android/AndroidControllers.java [hunk 1]
```diff
@@ -277,22 +277,27 @@ private void gatherControllers(boolean sendEvent) {
...
+					eventQueue.add(event);
+				}
+			} else {
+				controllers.add(controller);
 			}
-		} else {
-			controllers.add(controller);
+			Gdx.app.log(TAG, "added controller '" + name + "'");
+		} catch (RuntimeException e) {
+			// this exception is sometimes thrown by getDevice().
```
Comment (intrigus) [inline extensions/gdx-controllers/gdx-controllers-android/src/com/badlogic/gdx/controllers/android/AndroidControllers.java]: Wouldn't it be a good idea to at least print the exception?

**Comment by intrigus** [comment]
Do you have a stackstrace for a crash?

**Comment by MrStahlfelge** [comment]
See linked issue for stacktrace

**Comment by Tom-Ski** [comment]
Please update PR addressing @intrigus's review.

**Comment by MrStahlfelge** [comment]
I've added a log output when the exception is caught.

**Pull request merged**